### ISSUE TITLE
[Refactor] Simplify searchShortLink function

### ISSUE THREAD

**Issue by arberiii** [short-d/short#866]
**What is frustrating you?**
Function searchShortLink is hard to read, especially the for loop.
https://github.com/short-d/short/blob/284d1d6ae7c59549409860bda7c00d3abe9dd703/backend/app/usecase/search/search.go#L68
**Your solution**
Refactor this function appropriately.

**Comment by hemachandarv** [comment]
@arberiii Would this be a `good first issue`? I'd like to work on this, if it hasn't been done already.

**Comment by arberiii** [comment]
Yes, you can work on this! Please let me know if you have any trouble with the setup or anything else.

**Comment by hemachandarv** [comment]
Cool, I'll start working!
